xgcm/grid.py

```diff
@@ -1400,11 +1400,18 @@ def get_metric(self, array, axes, boundary=None, fill_value=None):
                         break
                 except KeyError:
                     pass
-        if metric_vars is None:
+        if metric_vars is not None:
+            # final check for metric_vars, where dimensions should match with input array
+            if set(metric_vars.dims).issubset(array_dims):
```

Comment:
does this ever get triggered or are you just making sure while you develop this feature?

Reply:
Under test_metric_ops_multi_axis/test_missingdim, tracer is averaged over `xt` and so array.dims are `zt` and `yt`, and then it tries to get the average over `"X"`. As it goes through get_metric, the resulting metric has dimensions `xu` and `yt`, and this happens because it goes to Condition 2. If I don't add this final check, that's the final metric that gets passed on to the test and the KeyError message under pytest fails.

Reply:
To clarify, using get_metric goes to Condition 2 because there are existing X-axis metrics specified. So basically, it finds metrics under the X-axis (`len overlap_metrics > 0`), then doesn't find a metric with a matching dimension (because `array.dims = zt and yt`)so it goes on to Condition 2 (interpolate metric with matching axis to desired dimensions).

Reply:
I am not sure I totally understand what is going on here. Do you have a short code snippet that illustrates the problem (e.g. a notebook cell I can paste)?

Reply:
```python
ds, coords, metrics = datasets_grid_metric("C)
grid = Grid(ds, coords=coords, metrics=metrics, periodic=periodic)
da = ds.tracer.mean("X)
axis = ("X")
weight = grid.get_metric(da, axis) # use current version of grid_metric with interp_like included
```

Reply:
The line `weight = grid.get_metric(da,axis)` is from average:

```python
def average(self, da, axis, **kwargs):
        
        weight = self.get_metric(da, axis)
        weighted = da.weighted(weight)

        # get dimension(s) corresponding
        # to `da` and `axis` input
        dim = self._get_dims_from_axis(da, axis)
        return weighted.mean(dim, **kwargs)
```
so in `test_metric_ops_multi_axis/test_missingdim`, we shouldn't be able to do `cumint, average or integrate` for cases where you don't have a metric that matches the exact dimensions of your DataArray `da`.